File: 0930-binary-subarrays-with-sum/0930-binary-subarrays-with-sum.py
class Solution(object):
    def numSubarraysWithSum(self, nums, goal):
        mp = {0: 1}
        ans, _sum = 0, 0
        for it in nums:
            _sum += it
            ans += mp.get(_sum - goal, 0)
            mp[_sum] = mp.get(_sum, 0) + 1
        return ans

File: 0930-binary-subarrays-with-sum/0930-binary-subarrays-with-sum.java
class Solution {
    public int numSubarraysWithSum(int[] nums, int goal) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        mp.put(0, 1);
        int ans = 0, sum = 0;
        for(int it : nums) {
            sum += it;
            ans += mp.getOrDefault(sum - goal, 0);
            mp.put(sum, mp.getOrDefault(sum, 0) + 1);
        }

        return ans;
    }
}